src/main/scala/viper/gobra/frontend/Config.scala

```diff
@@ -864,8 +872,8 @@ class ScallopGobraConfig(arguments: Seq[String], isInputOptional: Boolean = fals
   conflicts(directory, List(inclPackages, exclPackages))
 
   // must be lazy to guarantee that this value is computed only during the CLI options validation and not before.
-  lazy val isSiliconBasedBackend = backend.toOption match {
-    case Some(ViperBackends.SiliconBackend | _: ViperBackends.ViperServerWithSilicon) => true
+  lazy val isSiliconBasedBackend = backend.toOption.getOrElse(ConfigDefaults.DefaultBackend) match {
+    case ViperBackends.SiliconBackend | _: ViperBackends.ViperServerWithSilicon => true
```

Comment:
could we use `backendOrDefault` here instead of `backend.toOption.getOrElse(ConfigDefaults.DefaultBackend)`

Reply:
might be worth making this a def too

Reply:
`backend` in this context does not refer to `Config`'s field but the Scallop CLI option. Thus, calling `backendOrDefault` is not an option. However, I'm changing this to be a (private) `def`